### DISCUSSION TITLE
Draft GPR association

### DISCUSSION THREAD

**Discussion by sysyboom** [SysBioChalmers/RAVEN#501] [Help me!] [answered]
hi , I have two questions about RAVEN2.0.
1. I am trying to reconstruct a draft model by the function KEGGHomology=getKEGGModelForOrganism('KEGGHMMs','protein.fasta','euk90_kegg105','',false,false). After running this function,there are something confuse me that the genes of genes association  list are all connected with 'or' without even one 'and'. In fact, I think it should include two different Boolean logics, 'and' and 'or'.
2. When I use the function predictlocalization, the error 'W should contain at least one positive value' appears at function randomsample. I checked the GSS.score in the normalized GSS file and each row has at least one positive 1. I don't understand where else it might have gone wrong.

Looking forward to your reply!

**Comment by edkerk** [comment, marked as answer]
Hi @sysyboom, great that you're using RAVEN:

1. You're absolutely right that various gene associations should also include 'and'. Unfortunately, KEGG does not provide this. KEGG is built so that genes are assigned to KOs (KEGG ontologies), and these KOs are then assigned to reactions. From this we cannot know what is an essential subunit or what is an isoenzyme. Now I just noticed that KEGG has something called "Modules" as wel, which could maybe be explored, but this will not be straightforward to simply include. So while it is not possible now to define "and" from KEGG, this might be possible in the future.

2. I have little experience with this function, but when debugging pay attention that the W input is not just the GSS.scores as from the input file. During the run of `predictLocalization` it modifies `GSS.scores` multipe times, but probably more important is that the W input is `max(GSS.scores(I,:),[],2)-GSS.scores(sub2ind(size(g2c),I,J))+0.1`. If you run `dbstop if error`, you can check at that moment in function why this expression does not give a positive value.

**Reply by sysyboom** [reply]
thanks for your reply.I will try it again!
